### PR TITLE
feat(pack.move): retrieve buildings on move offer

### PR DESCRIPTION
**Pull request by Steffy29**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #UXCT-321
| License          | BSD 3-Clause

## Description

- Rework on eligibility test for eligibility by line and by address
- On the building details, display the building selected, not possible to change the building reference

### CODE REVIEW

File: packages/manager/apps/telecom/src/app/telecom/pack/move/building-details/move-building-details.controller.js [hunk 1]
```diff
@@ -31,33 +31,31 @@ export default class MoveBuildingDetailsCtrl {
...
-        this.buildings[i].buildingName = this.$translate.instant(
-          'pack_move_building_details_unknown',
-        );
-      }
+    // check if the building name is empty to set a name to display in the select component
+    if (this.building.name === '') {
+      this.building.name = this.$translate.instant(
+        'pack_move_building_details_unknown',
+      );
+    }
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/move/building-details/move-building-details.controller.js:39]: nitpicking:
```suggestion
this.building.name ||= this.$translate.instant(
        'pack_move_building_details_unknown',
      );
```

File: packages/manager/apps/telecom/src/app/telecom/pack/move/building-details/move-building-details.controller.js [hunk 2]
```diff
@@ -31,33 +31,31 @@ export default class MoveBuildingDetailsCtrl {
...
-    });
+    this.OvhApiConnectivityEligibilitySearch.v6()
+      .buildingDetails({}, params)
+      .$promise.then((buildingDetails) => {
+        if (has(buildingDetails, 'result.stairs')) {
+          this.building.stairs = buildingDetails.result.stairs.map((stair) =>
+            this.convertStairs(stair),
+          );
+        }
+      })
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/move/building-details/move-building-details.controller.js:53]: no `catch` ?

File: packages/manager/apps/telecom/src/app/telecom/pack/move/building-details/move-building-details.html [hunk 3]
```diff
@@ -10,18 +10,7 @@ <h3 data-translate="pack_move_building_details_title"></h3>
...
-                data-title="{{:: 'pack_move_building_details_select_a_building' | translate}}"
-                placeholder="{{:: 'pack_move_building_details_select_a_building' | translate}}"
-                data-items="$ctrl.buildings"
-                required
-                data-match="buildingName"
-                data-on-change="$ctrl.changeSelection()"
-            >
-                <span data-ng-bind="$item.buildingName"></span>
-            </oui-select>
+            <div data-ng-bind="$ctrl.model.selectedBuilding.name}}"></div>
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/move/building-details/move-building-details.html:13]:
```suggestion
            <p data-ng-bind="$ctrl.model.selectedBuilding.name}}"></p>
```

File: packages/manager/apps/telecom/src/app/telecom/pack/move/eligibility/address/move-eligibility-address.controller.js [hunk 4]
```diff
@@ -177,30 +155,17 @@ export default class {
...
-              .testBuilding(this.$scope, {
-                building: building.reference,
-                index,
-              })
-              .then((elig) => {
-                return buildingsEligible.push(elig);
-              });
-          });
-          return buildingsEligible;
+          return data.result;
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/move/eligibility/address/move-eligibility-address.controller.js:168]: nitpicking:
```suggestion
          return data.result.length > 0 ? data.result : null
```

File: packages/manager/apps/telecom/src/app/telecom/pack/move/eligibility/lineNumber/move-eligibility-lineNumber.controller.js [hunk 5]
```diff
@@ -4,108 +4,52 @@ export default class {
...
-                offer.endpoint.referenceTypeFiber =
-                  fiber.result.endpoint.referenceType;
-              }
-              offer.buildingReference =
-                fiber.result.endpoint.fiberInfo.buildingReference;
-            }
-            return offer;
-          },
-        );
+        return copper;
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/move/eligibility/lineNumber/move-eligibility-lineNumber.controller.js:39]: The `then` is not needed if you just return the value

File: packages/manager/apps/telecom/src/app/telecom/pack/move/eligibility/lineNumber/move-eligibility-lineNumber.controller.js [hunk 6]
```diff
@@ -115,42 +59,176 @@ export default class {
       });
   }
 
+  fiberEligibilityByBuilding(buildingRef) {
+    return this.OvhApiConnectivityEligibility.v6()
+      .testBuilding(this.$scope, {
+        building: buildingRef,
+      })
+      .then((elig) => {
+        console.log('building', elig);
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/move/eligibility/lineNumber/move-eligibility-lineNumber.controller.js:68]:
```suggestion

```
same here, then is not needed if you return the same value

File: packages/manager/apps/telecom/src/app/telecom/pack/move/eligibility/lineNumber/move-eligibility-lineNumber.html [hunk 7]
```diff
@@ -33,12 +33,154 @@
...
+            data-translate="submit"
+            data-ng-disabled="$ctrl.loading || availabilityForm.phoneNumber.$invalid || $ctrl.method !== 'number'"
+        ></button>
+        <oui-spinner class="ml-2" data-ng-if="$ctrl.loading" data-size="s">
+        </oui-spinner>
+    </div>
+
+    <div data-ng-if="$ctrl.displayFirstStep" class="mt-3">
+        <div class="row">
+            <div class="col-md-6" data-ng-repeat="copper in $ctrl.copperOffers">
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/move/eligibility/lineNumber/move-eligibility-lineNumber.html:49]:
```suggestion
            <div class="col-md-6" data-ng-repeat="copper in $ctrl.copperOffers track by copper.id">
```

File: packages/manager/apps/telecom/src/app/telecom/pack/move/eligibility/lineNumber/move-eligibility-lineNumber.html [hunk 8]
```diff
@@ -33,12 +33,154 @@
...
+                        data-ng-if="copper.status === $ctrl.STATUS.active"
+                        data-translate="pack_move_eligibility_line_status_active"
+                    ></h4>
+                    <h4
+                        class="oui-box__heading"
+                        data-ng-if="copper.status === $ctrl.STATUS.inactive"
+                        data-translate="pack_move_eligibility_line_status_inactive"
+                    ></h4>
+
+                    <div data-ng-if="copper.errorMessage">
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/move/eligibility/lineNumber/move-eligibility-lineNumber.html:62]: No `oui-message` for this ?

File: packages/manager/apps/telecom/src/app/telecom/pack/move/eligibility/lineNumber/move-eligibility-lineNumber.html [hunk 9]
```diff
@@ -33,12 +33,154 @@
...
+                            </div>
+                            <div>
+                                <span
+                                    data-ng-bind="copper.endpoint.address.zipCode"
+                                ></span>
+                                <span
+                                    data-ng-bind="copper.endpoint.address.city"
+                                ></span>
+                            </div>
+                        </div>
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/move/eligibility/lineNumber/move-eligibility-lineNumber.html:90]: Maybe we can use the `address` tag in this case https://developer.mozilla.org/en-US/docs/Web/HTML/Element/address

File: packages/manager/apps/telecom/src/app/telecom/pack/move/resume/move-resume.controller.js [hunk 10]
```diff
@@ -262,6 +264,64 @@ export default class MoveResumeCtrl {
         assign(moveData, {
           meeting,
         });
+
+        // Set address complements for copper line creation if set
+        if (this.offer.selected.searchAddress.residence) {
+          assign(moveData, {
+            residence: this.offer.selected.searchAddress.residence,
+          });
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/move/resume/move-resume.controller.js:272]:
```suggestion
          Object.assign(moveData, {
            residence: this.offer.selected.searchAddress.residence,
          });
```
maybe we don't need lodash in this particular case

File: packages/manager/apps/telecom/src/app/telecom/pack/move/eligibility/lineNumber/move-eligibility-lineNumber.html (hunk 8 above)
Comment (Steffy29) [reply packages/manager/apps/telecom/src/app/telecom/pack/move/eligibility/lineNumber/move-eligibility-lineNumber.html:62]: No it's a message to inform user that there is an error on eligibility test on the line number like "No active line found".

**Review by antleblanc — APPROVED** [review]

**Pull request merged**